user_config.go

```diff
@@ -354,16 +354,16 @@ func convertTerraformUserConfigValueToAPICompatibleFormat(
 			omit = true
 		}
 	case "number":
-		switch value.(type) {
+		switch res := value.(type) {
 		case float64:
 		case float32:
```

Comment:
You could use `case float32, int64, int32, int:` here (I think) to avoid repeating `convertedValue = float64(res)` on every separate case. Not sure whether that would be better, though.

Reply:
Doesn't work, it causes `res` to be treated as `interface{}`